photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/estimators/GaussianProcessEstimator.scala

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.hyperparameter.estimators
+
+import breeze.linalg.{DenseMatrix, DenseVector, cholesky, diag, sum}
+import breeze.numerics.constants.Pi
+import breeze.numerics.log
+import breeze.stats.mean
+
+import com.linkedin.photon.ml.hyperparameter.Linalg.choleskySolve
+import com.linkedin.photon.ml.hyperparameter.SliceSampler
+import com.linkedin.photon.ml.hyperparameter.estimators.kernels._
+
+/**
+ * Estimates a Gaussian Process regression model
+ *
+ * @see "Gaussian Processes for Machine Learning" (GPML), http://www.gaussianprocess.org/gpml/, Chapter 2
+ *
+ * @param kernel the covariance kernel
+ * @param normalizeLabels if true, the estimator normalizes labels to a mean of zero before fitting
+ * @param predictionTransformation tranformation function to apply for predictions
+ * @param monteCarloNumBurnInSamples the number of samples to draw during the burn-in phase of kernel parameter
+ *   estimation
+ * @param monteCarloNumSamples the number of samples to draw for estimating kernel parameters
+ */
+class GaussianProcessEstimator(
+    kernel: Kernel = new RBF,
+    normalizeLabels: Boolean = false,
+    predictionTransformation: Option[PredictionTransformation] = None,
+    monteCarloNumBurnInSamples: Int = 100,
+    monteCarloNumSamples: Int = 100,
```

Comment:
They appeared to be good rules of thumb during implementation / experimentation. It's not super important to get absolutely correct settings here, since a single sample is roughly equivalent to MLE (in fact, the spearment implementation just takes a single sample).